rules/aip0233/message_rules.go

```diff
@@ -0,0 +1,125 @@
+
+import (
+	"fmt"
+
+	"github.com/gertd/go-pluralize"
+	"github.com/googleapis/api-linter/lint"
+	"github.com/jhump/protoreflect/desc"
+	"github.com/jhump/protoreflect/desc/builder"
+)
+
+// The Batch Create request message should have parent field.
+var parentField = &lint.MessageRule{
+	Name:   lint.NewRuleName("core", "0233", "request-message", "parent-field"),
+	URI:    "https://aip.dev/233#request-message",
+	OnlyIf: isBatchCreateRequestMessage,
+	LintMessage: func(m *desc.MessageDescriptor) []lint.Problem {
+		// Rule check: Establish that a `parent` field is present.
+		parentField := m.FindFieldByName("parent")
+		if parentField == nil {
+			return []lint.Problem{{
+				Message:    fmt.Sprintf(`Message %q has no "parent" field`, m.GetName()),
+				Descriptor: m,
+			}}
+		}
+
+		// Rule check: Establish that the parent field is a string.
+		if parentField.GetType() != builder.FieldTypeString().GetType() {
+			return []lint.Problem{{
+				Message:    `"parent" field on create request message must be a string`,
+				Descriptor: parentField,
+			}}
+		}
+
+		return nil
+	},
+}
+
+// The Batch Create standard method should have repeated standard create request
+// message field.
+var requestsField = &lint.MessageRule{
+	Name:   lint.NewRuleName("core", "0233", "request-message", "name-field"),
+	URI:    "https://aip.dev/233#request-message",
+	OnlyIf: isBatchCreateRequestMessage,
+	LintMessage: func(m *desc.MessageDescriptor) (problems []lint.Problem) {
+		// Rule check: Establish that a "requests" field is present.
+		requests := m.FindFieldByName("requests")
+
+		// Rule check: Ensure that the "requests" field is existed.
+		if requests == nil {
+			return []lint.Problem{{
+				Message:    fmt.Sprintf(`Message %q has no "requests" field`, m.GetName()),
+				Descriptor: m,
+			}}
+		}
+
+		// Rule check: Ensure that the standard create request message field "requests" is repeated.
+		if !requests.IsRepeated() {
+			problems = append(problems, lint.Problem{
+				Message:    `The "requests" field should be repeated`,
+				Descriptor: requests,
+			})
+		}
+
+		// Rule check: Ensure that the standard create request message field is the
+		// correct type. Note: Use m.GetName()[11:len(m.GetName())-7]) to retrieve
+		// the resource name from the the batch create request, for example:
+		// "BatchCreateBooksRequest" -> "Books"
+		rightTypeName := fmt.Sprintf("Create%sRequest", pluralize.NewClient().Singular(m.GetName()[11:len(m.GetName())-7]))
+		if requests.GetMessageType() == nil || requests.GetMessageType().GetName() != rightTypeName {
+			problems = append(problems, lint.Problem{
+				Message:    fmt.Sprintf(`The "requests" field on Batch Create Request should be a %q type`, rightTypeName),
+				Descriptor: requests,
+			})
+		}
+		return
+	},
+}
+
+// The Batch Create response message should have resource field.
+var resourceField = &lint.MessageRule{
+	Name:   lint.NewRuleName("core", "0233", "response-message", "resource-field"),
```

Comment:
Can you reduce the length of the rule name to three parts?

Reply:
done.